kong/router/atc_compat.lua

```diff
@@ -421,11 +422,16 @@ local function add_atc_matcher(inst, route, route_id,
 
   end
 
-  if remove_existing then
-    assert(inst:remove_matcher(route_id))
+  if remove_existing and not inst:remove_matcher(route_id) then
```

Comment:
Same, we can keep the `assert` for `remove_matcher`.

Reply:
done